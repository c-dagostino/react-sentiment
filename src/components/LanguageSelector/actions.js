export const GET_LOCALE = 'GET_LOCALE';
export const SET_LOCALE = 'SET_LOCALE';

export function getLocale() {
  return {
    type: GET_LOCALE
  };
}

export function setLocale(locale) {
  return {
    type: SET_LOCALE,
    locale
  };
}
