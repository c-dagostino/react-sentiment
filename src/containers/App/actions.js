import { LOAD_REPOS_ERROR } from './constants';

export function repoLoadingError(error) {
  return {
    type: LOAD_REPOS_ERROR,
    error
  };
}
