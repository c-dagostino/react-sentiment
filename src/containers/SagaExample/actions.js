export const GET_SAGA_TEST = 'SAGA_TEST';
export const SET_SAGA_TEST = 'SET_SAGA_TEST';

export function getSagaTest() {
  return {
    type: GET_SAGA_TEST
  };
}

export function setSagaTest(val) {
  return {
    type: SET_SAGA_TEST,
    test: val
  };
}
