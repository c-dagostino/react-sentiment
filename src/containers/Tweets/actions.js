import Log from '../../utils/Log';
import { FETCH_TWEETS, FETCH_TWEETS_SUCCESS } from '../App/constants';

export function fetchTweets(searchTerm) {
  Log.trace('fetchTweets', 'Tweets actions');

  return {
    type: FETCH_TWEETS,
    searchTerm
  };
}

export function fetchTweetsSuccess(tweets) {
  Log.trace('fetchTweetsSuccess', 'Tweets actions');
  return {
    type: FETCH_TWEETS_SUCCESS,
    tweets
  };
}
