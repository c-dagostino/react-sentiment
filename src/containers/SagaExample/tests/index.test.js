import React from 'react';
import { fromJS } from 'immutable';
import { shallow } from 'enzyme';

import { SagaPage, mapStateToProps, mapDispatchToProps } from '../index';

import * as actions from '../actions';

function getWrapper(props) {
  const requiredProps = {
    getSagaTest: () => {}
  };

  return shallow(<SagaPage {...requiredProps} {...props} />);
}

describe('SagaPage', () => {
  it('Expect component to render properly', () => {
    const wrapper = getWrapper();
    expect(wrapper).toMatchSnapshot();
  });

  it('Expect button to dispatch getSagaTest action', () => {
    const props = {
      getSagaTest: jest.fn()
    };
    const wrapper = getWrapper(props);
    wrapper.find('[data-id="sagaTestButton"]').simulate('click', {
      preventDefault: () => {}
    });
    expect(props.getSagaTest).toHaveBeenCalled();
  });
});

describe('SagaPage mapStateToProps', () => {
  let state = {};
  beforeEach(() => {
    const saga = {
      test: 'saga-test'
    };

    state = fromJS(saga);
  });

  it('Expect sagaTest to be populated from state', () => {
    const result = mapStateToProps(state).sagaTest;
    expect(result).toBe(state.getIn(['saga', 'test']));
  });
});

describe('SagaPage mapDispatchToProps', () => {
  it('Expect mapDispatchToProps to dispatch getSagaTest', () => {
    const mockAction = jest.spyOn(actions, 'getSagaTest');
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).getSagaTest();
    expect(mockAction).toHaveBeenCalled();
  });
});
