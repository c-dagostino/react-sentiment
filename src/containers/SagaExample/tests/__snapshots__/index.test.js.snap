// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SagaPage Expect component to render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SagaPage
    getSagaTest={[Function]}
    sagaTest=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Saga Example
</h3>,
        <p>
          Test: 
</p>,
        <button
          data-id="sagaTestButton"
          onClick={[Function]}
          type="button"
>
          Test Saga
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Saga Example",
        },
        "ref": null,
        "rendered": "Saga Example",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test: ",
        },
        "ref": null,
        "rendered": "Test: ",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test Saga",
          "data-id": "sagaTestButton",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Test Saga",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Saga Example
</h3>,
          <p>
            Test: 
</p>,
          <button
            data-id="sagaTestButton"
            onClick={[Function]}
            type="button"
>
            Test Saga
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Saga Example",
          },
          "ref": null,
          "rendered": "Saga Example",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test: ",
          },
          "ref": null,
          "rendered": "Test: ",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test Saga",
            "data-id": "sagaTestButton",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Test Saga",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
